### DISCUSSION TITLE
Using Remix with vanilla.js

### DISCUSSION THREAD

**Discussion by emrekardaslar** [remix-run/remix#7209] [Q&A] [answered]
I want to add a banner on the  top of my screen and I want to do that on the client-side. I also want it to load immediately since I don't want CLS on my site. Vanilla.js script can place it immediately, much faster than useEffect but after the render the banner disappears. How can I use my vanilla.js without being affected from React or Remix?

**Comment by sergiodxa** [comment]
You can remove the Scripts component, but you will not have client-side React anymore, you can still use client-side JS code by loading it from a `.js` file saved on the public folder or inlining it.

But why not render it server-side? Is there no way you can know you need to render that banner on the loader? That way you could send the HTML directly with the banner.

**Reply by emrekardaslar** [reply]
The system uses cloudflare as cache. So this banner should be rendered on the client as fast as possible. Normally if you use vanilla.js without react it can load immidately. I am asking that is there an equivalent thing for that in React or Remix?

**Reply by emrekardaslar** [reply, marked as answer]
When I had rendered whole header using dangerousInnerHtml and moved the function calls in useEffect to the parent I managed to do that.
